examples/persistence/badger/main.go

```diff
@@ -31,8 +34,20 @@ func main() {
 	server := mqtt.New(nil)
 	_ = server.AddHook(new(auth.AllowHook), nil)
 
+	// AddHook adds a BadgerDB hook to the server with the specified options.
+	// GcInterval specifies the interval at which BadgerDB garbage collection process runs.
+	// Refer to https://dgraph.io/docs/badger/get-started/#garbage-collection for more information.
 	err := server.AddHook(new(badger.Hook), &badger.Options{
-		Path: badgerPath,
+		Path:       badgerPath,
+		GcInterval: 3 * time.Minute, // Set the interval for garbage collection.
```

Comment:
Have you tested this with a shorter `GcInterval` and the impact that has on performance? If the GC functions as a "stop the world" type and with higher client counts it may be a bit too high.

Reply:
As far as I know, Badger GC would not reclaim the latest value log, which is a read-write log. It only works on the value logs that have become "immutable". So there should be no STW (Stop-The-World) issue.

GC certainly comes with a cost. According to the Badger official documentation: "It is recommended to be called during periods of low activity in your system, or periodically." Regarding the GcInterval, we are just providing an example for reference here. As for GcInterval or other options, users may need to optimize based on their actual scenarios.